wrapanapi/systems/msazure.py

```diff
@@ -794,24 +812,38 @@ def remove_pips_by_search(self, pip_template, resource_group=None):
             self.logger.info('"%s" pip removed', pip)
             results.append((pip, operation.status()))
         if not results:
-            self.logger.debug('No PIPs matching "%s" template were found', pip_template)
+            self.logger.debug('No unused/unattached PIPs found to be removed!')
         return results
 
-    def remove_discs_by_search(self, disc_name):
+    def remove_discs_by_search(self, disc_name=None, resource_group=None):
         """
-        Used for clean_up jobs to remove discs that are left after deleting the VM
+        Used for clean_up jobs to remove disc(s) that are left after deleting the VM
         """
-        self.logger.info('Attempting to find the disc image {}'.format(disc_name))
         results = []
-        discs = self.find_templates(container='system',
-                                    prefix='{}{}'.format(self.orphaned_discs_path, disc_name))
-
-        for disc in discs:
-            disc.delete()
-            self.logger.info('disc {} removed'.format(disc_name))
-            results.append(disc_name)
-        if not results:
-            self.logger.debug('No discs matching {} were found'.format(disc_name))
+        if disc_name:
+            self.logger.info('Attempting to find the disc image {}'.format(disc_name))
+            discs = self.find_templates(container='system',
+                                        prefix='{}{}'.format(self.orphaned_discs_path, disc_name))
+
+            for disc in discs:
+                disc.delete()
+                self.logger.info('disc {} removed'.format(disc_name))
+                results.append(disc_name)
+            if not results:
+                self.logger.debug('No discs matching {} were found'.format(disc_name))
+        else:
+            # Remove all discs
+            self.logger.info('Attempting to find all the unattached disks and delete.')
+            discs = self.list_free_discs(resource_group=self.resource_group)
+            for disc_name in discs:
+                operation = self.compute_client.disks.delete(
+                    resource_group_name=resource_group or self.resource_group,
+                    disk_name=disc_name)
+                operation.wait()
+                self.logger.info('"%s" disc removed', disc_name)
+                results.append((disc_name, operation.status()))
+            if not results:
+                self.logger.debug('No unused discs matching {} were found')
```

Comment:
Missing f-string context? `{}` with no value to be formatted

Reply:
Nop it was left to be removed as its a general mssg.